### DISCUSSION TITLE
Easy indexes with Migrations

### DISCUSSION THREAD

**Discussion by MoreBlood** [directus/directus#16969] [Feature Requests]
During some perfomance testing, we've found out low directus generated db prefomance
As an easy and logical fix was adding indexes

Here is small code snippet for [migration ](https://docs.directus.io/extensions/migrations.html)

```js
module.exports = {
  async up(knex) {
    await knex.schema.table('user', async function (table) {
      table.index(['id'], 'user_id', {
        indexType: 'hash',
      });
      table.index(['country'], 'user_country', {
        indexType: 'btree',
      });
      table.index(['spa'], 'user_spa', {
        indexType: 'btree',
      });
      table.index(['country', 'spa'], 'user_country_spa', {
        indexType: 'btree',
      });
      table.index(['username'], 'user_username', {
        indexType: 'btree',
      });
    });
  },
};
```

Hope it will help someone!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
